gtk4/src/string_object.rs

```diff
@@ -0,0 +1,59 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::StringObject;
+use glib::{GStr, GString};
+
+impl From<StringObject> for String {
+    #[inline]
+    fn from(value: StringObject) -> Self {
+        value.string().to_string()
+    }
+}
+
+impl From<StringObject> for GString {
+    #[inline]
+    fn from(value: StringObject) -> Self {
+        value.string()
+    }
+}
+
+impl From<String> for StringObject {
```

Comment:
```suggestion
impl<'a> From<&String> for StringObject {
```

in addition (and also for `GString`). It's usually covered by the `&str` / `&GStr` impls but not in every situation.